tests/cases/fourslash/quickInfoJsDocTags12.ts

```diff
@@ -0,0 +1,17 @@
+/// <reference path="fourslash.ts" />
+
+/////**
+//// * @param a - a parameter
+//// * @param b - b parameter
+//// * @param c - c parameter
+//// * @param f - f parameter
```

Comment:
getJSDocParameterTagWorker also supports matching binding patterns by index. Does it work to mix the two modes? If so, please add a test. Here's what I got:

```ts
/**
 * @param a - a pram
 * @param bindingelement - for { b, c }
 * @param b - b parm
 * @param bindingelement2 - for { e } 
 * @param c - c param
 * @param d - d param
 */
function foo(a: string, { b, c }: { b: string, c: string }, d: string, { e }: { e: string }) {
  return a + b + c + d + e
}
```

Reply:
Do we need to omit matching binding patterns by index?

```
/////**
//// * @param a - a param
//// * @param bindingelement1 - for { b, c }
//// * @param b - b param
//// * @param bindingelement2 - for { e }
//// * @param c - c param
//// * @param d - d param
//// */
////function foo(a: string, { b, c }: { b: string, c: string }, d: string, { e }: { e: string }) {
////    a/*1*/
////    b/*2*/
////    c/*3*/
////    d/*4*/
////    e/*5*/
////}
```

<details>
  <summary>quickInfoJsDocTags13.baseline</summary>

```json
[
  {
    "marker": {
      "fileName": "/tests/cases/fourslash/quickInfoJsDocTags13.ts",
      "position": 275,
      "name": "1"
    },
    "quickInfo": {
      "kind": "parameter",
      "kindModifiers": "",
      "textSpan": {
        "start": 274,
        "length": 1
      },
      "displayParts": [
        {
          "text": "(",
          "kind": "punctuation"
        },
        {
          "text": "parameter",
          "kind": "text"
        },
        {
          "text": ")",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "a",
          "kind": "parameterName"
        },
        {
          "text": ":",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "string",
          "kind": "keyword"
        }
      ],
      "documentation": [
        {
          "text": "- a param",
          "kind": "text"
        }
      ],
      "tags": [
        {
          "name": "param",
          "text": [
            {
              "text": "a",
              "kind": "parameterName"
            },
            {
              "text": " ",
              "kind": "space"
            },
            {
              "text": "- a param",
              "kind": "text"
            }
          ]
        }
      ]
    }
  },
  {
    "marker": {
      "fileName": "/tests/cases/fourslash/quickInfoJsDocTags13.ts",
      "position": 281,
      "name": "2"
    },
    "quickInfo": {
      "kind": "parameter",
      "kindModifiers": "",
      "textSpan": {
        "start": 280,
        "length": 1
      },
      "displayParts": [
        {
          "text": "(",
          "kind": "punctuation"
        },
        {
          "text": "parameter",
          "kind": "text"
        },
        {
          "text": ")",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "b",
          "kind": "parameterName"
        },
        {
          "text": ":",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "string",
          "kind": "keyword"
        }
      ],
      "documentation": [
        {
          "text": "- b param",
          "kind": "text"
        }
      ],
      "tags": [
        {
          "name": "param",
          "text": [
            {
              "text": "b",
              "kind": "parameterName"
            },
            {
              "text": " ",
              "kind": "space"
            },
            {
              "text": "- b param",
              "kind": "text"
            }
          ]
        }
      ]
    }
  },
  {
    "marker": {
      "fileName": "/tests/cases/fourslash/quickInfoJsDocTags13.ts",
      "position": 287,
      "name": "3"
    },
    "quickInfo": {
      "kind": "parameter",
      "kindModifiers": "",
      "textSpan": {
        "start": 286,
        "length": 1
      },
      "displayParts": [
        {
          "text": "(",
          "kind": "punctuation"
        },
        {
          "text": "parameter",
          "kind": "text"
        },
        {
          "text": ")",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "c",
          "kind": "parameterName"
        },
        {
          "text": ":",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "string",
          "kind": "keyword"
        }
      ],
      "documentation": [
        {
          "text": "- c param",
          "kind": "text"
        }
      ],
      "tags": [
        {
          "name": "param",
          "text": [
            {
              "text": "c",
              "kind": "parameterName"
            },
            {
              "text": " ",
              "kind": "space"
            },
            {
              "text": "- c param",
              "kind": "text"
            }
          ]
        }
      ]
    }
  },
  {
    "marker": {
      "fileName": "/tests/cases/fourslash/quickInfoJsDocTags13.ts",
      "position": 293,
      "name": "4"
    },
    "quickInfo": {
      "kind": "parameter",
      "kindModifiers": "",
      "textSpan": {
        "start": 292,
        "length": 1
      },
      "displayParts": [
        {
          "text": "(",
          "kind": "punctuation"
        },
        {
          "text": "parameter",
          "kind": "text"
        },
        {
          "text": ")",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "d",
          "kind": "parameterName"
        },
        {
          "text": ":",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "string",
          "kind": "keyword"
        }
      ],
      "documentation": [
        {
          "text": "- d param",
          "kind": "text"
        }
      ],
      "tags": [
        {
          "name": "param",
          "text": [
            {
              "text": "d",
              "kind": "parameterName"
            },
            {
              "text": " ",
              "kind": "space"
            },
            {
              "text": "- d param",
              "kind": "text"
            }
          ]
        }
      ]
    }
  },
  {
    "marker": {
      "fileName": "/tests/cases/fourslash/quickInfoJsDocTags13.ts",
      "position": 299,
      "name": "5"
    },
    "quickInfo": {
      "kind": "parameter",
      "kindModifiers": "",
      "textSpan": {
        "start": 298,
        "length": 1
      },
      "displayParts": [
        {
          "text": "(",
          "kind": "punctuation"
        },
        {
          "text": "parameter",
          "kind": "text"
        },
        {
          "text": ")",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "e",
          "kind": "parameterName"
        },
        {
          "text": ":",
          "kind": "punctuation"
        },
        {
          "text": " ",
          "kind": "space"
        },
        {
          "text": "string",
          "kind": "keyword"
        }
      ],
      "documentation": []
    }
  }
]
```
</details>